### PR TITLE
Link manifst docs to version info in crates.io docs

### LINKED ISSUE #1707: docs show example of `version = "*"` w/o explanation
I was wondering what the precise semantics of using `version = "*"` were, especially what the "gotchas" are for using it.

So I skimmed over the cargo docs at http://doc.crates.io/manifest.html but could not find any discussion of what `version = "*"` even *means*.  The docs do say that:

> Cargo bakes in the concept of [Semantic Versioning](http://semver.org/)

but that link goes to http://semver.org/, which does not include any discussion of something like `"*"`.

I ended up resorting to looking at the cargo source code, which led me to http://doc.rust-lang.org/semver/semver/index.html which has a section labelled "Wildcard requirements" that has the most important piece of information I was looking for (namely, what `"*"` *means*.)

So, two things to fix:
 1. Say what `version = "*"` means
 2. Add some discussion about when it is appropriate to use `version = "*"`, versus when one should put in a specific version label.

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #1707

### CODE REVIEW

File: src/doc/manifest.md [hunk 1]
```diff
@@ -24,6 +24,9 @@ basic rules:
   `pub` structs, traits, fields, types, functions, methods or anything else.
 * Use version numbers with three numeric parts such as 1.0.0 rather than 1.0.
 
+For more on versions, see [this
+documentation][http://doc.crates.io/crates-io.html#using-crates.io-based-crates].
```
Comment (alexcrichton) [inline src/doc/manifest.md]: Could this use a relative URL?

**Comment by alexcrichton** [comment]
r=me with a relative URL

File: src/doc/manifest.md (hunk 1 above)
Comment (steveklabnik) [inline src/doc/manifest.md]: whoops! nice catch

**Pull request merged**